flexget/components/managed_lists/lists/ombi_list.py

```diff
@@ -0,0 +1,548 @@
+
+        response = self._post_data(api_endpoint, data)
+
+        if not response:
+            # Unable to mark this item as requested, but what now?
+            return
+
+        if response.get('isError'):
+            log.error(
+                f"Failed to mark {self.data['title']} as available because: {response['errorMessage']}"
+            )
+
+        log.verbose(f"{self.data['title']} has been marked available.")
+
+
+class OmbiMovie(OmbiEntry):
+    """Manage a Movie entry in Ombi."""
+
+    entry_type = 'movie'
+
+    def __init__(
+        self, base_url: str, auth: Callable[[], Dict[str, str]], data: Dict[str, Any]
+    ) -> None:
+        super().__init__(base_url, auth, self.entry_type, data)
+
+    @classmethod
+    def from_imdb_id(cls, base_url: str, tmdb_id: str, auth: Callable[[], Dict[str, str]]):
+        """Create a Ombi Entry from an IMDB ID."""
+
+        headers = {'Content-Type': 'application/json', 'Accept': 'application/json'}
+        headers.update(auth())
+
+        url = f"{base_url}/api/v2/Search/{cls.entry_type}/{tmdb_id}"
+
+        response = requests.get(url, headers=headers)
+
+        if not response.ok:
+            log.error(f"{url} return {response.reason}")
+            # Do I throw an exception here? Is that allowed from a plugin?
+            return None
+
+        data = response.json()
+
+        # Their is a bug in OMBI where if you get a movie by its TMDB ID
+        # then the theMovieDbId field will be blank for some reason...
+        data['theMovieDbId'] = tmdb_id
+
+        return OmbiMovie(base_url, auth, data)
+
+
+# In the future this might need split out into 3 classes (Show, Season, Episode)
+class OmbiTv(OmbiEntry):
+    """Manage a TV entry in Ombi."""
+
+    entry_type = 'tv'
+
+    def __init__(
+        self, base_url: str, auth: Callable[[], Dict[str, str]], data: Dict[str, Any]
+    ) -> None:
+        super().__init__(base_url, auth, self.entry_type, data)
+
+    @classmethod
+    def from_tvdb_id(cls, base_url: str, tmdb_id: str, auth: Callable[[], Dict[str, str]]):
+        """Create a Ombi Entry from an TVDB ID."""
+
+        headers = {'Content-Type': 'application/json', 'Accept': 'application/json'}
+        headers.update(auth())
+
+        url = f"{base_url}/api/v2/Search/{cls.entry_type}/moviedb/{tmdb_id}"
+
+        response = requests.get(url, headers=headers)
+
+        if not response.ok:
+            log.error(f"{url} return {response.reason}")
+            # Do I throw an exception here? Is that allowed from a plugin?
+            return None
+
+        data = response.json()
+
+        # Their is a bug in OMBI where if you get a movie by its TMDB ID
+        # then the theMovieDbId field will be blank for some reason...
+        data['theMovieDbId'] = tmdb_id
+
+        return OmbiTv(base_url, auth, data)
+
+
+class OmbiSet(MutableSet):
+    """The schema for the Ombi managed list."""
+
+    supported_ids = ['imdb_id', 'tmdb_id', 'ombi_id']
```

Comment:
We actually dont support all three of these. We only `tmdb_id` for TV stuff. I'm not sure what we support for movies yet. Either `imdb_id` or `tmdb`? 

But we don't support `ombi_id` currently. Which means adding things from one ombi list to another will fail if you don't use a lookup? But this should be fixable.

Reply:
@bagobones We should probably support the following:

Movies
- tmdb_id via `/api/v2/Search/movie/${tmdb_id}`
- imdb_id vi `/api/v2/Search/movie/${imdb_id}`

Tv
- tvdb_id via `/api/v2/Search/tv/{tvdbId}`

It would be nice to support `ombi_id` but I'm not even sure we can? I'm looking at the ombi API https://demo.ombi.io/swagger/index.html and I see that when you make a call like `/api/v2/Search/movie/${tmdb_id}` it retuns JSON like:
```JSON
{
  ...
 "id": 0,
  "approved": true,
  "denied": true,
  "deniedReason": "string",
  "requested": true,
  "requestId": 0,
  "available": true,
  "plexUrl": "string",
  "embyUrl": "string",
  "jellyfinUrl": "string",
  "quality": "string",
  "imdbId": "string",
  "theTvDbId": "string",
  "theMovieDbId": "string"
}
```
Where `id` is the `ombi_id`? I guess. But I don't see an API endpoint where we can pass the `ombi_id` and get back all the other data. Let me know if you find one. We could also just do a title search but that sounds prone to error.

I think we should modify the `class OmbiMovie(OmbiEntry):` to have 3 class methods. 

```python
def from_imdb_id() --> OmbiMovie:
    pass
def from_tmdb_id() --> OmbiMovie:
    pass
def from_id(cls, entry: Entry) --> OmbiMovie:
    if "imdb_id" in entry:
        return cls.from_imdb(cls, entry['imdb_id'])
    if "tmdb_id" in entry:
        return cls.from_tmdb(cls, entry['tmdb_id'])

    # Log an error and throw an exception?
```

Then we just need to update `_get_ombi_entry` to call `from_id`. Oh and update the supported_id's field. 

Not trying to micromanage you, take it as a suggestion. Tomorrow I'm going to work on getting OMBI setup so I can have a working development environment. Then I will be able to contribute. Lots came back after reading the code and the [list API docs](https://github.com/Flexget/Flexget/blob/develop/docs/develop/list_interface.rst).

Reply:
My memory is foggy as it has been YEARs, but I think that was yanked right from my original code, the idea at the time for supporting ombi ID was that if you ran a task taking ombi as an input and then actions on ombi as an output like deleting items or marking them as complete you would want to use the ombi ID not the movie / tv ID in case they did not exist.

Not sure if that is actually relevant as I think the movie or tv ID are always present.

Reply:
@bagobones The issue I am seeing, is that the only way to mark a movie as requested is to use the `tmdb_id`. Which is a problem if you are looking up a movie with the `imdb_id` because the returned data for the movie does not include the `tmdb_id`. I have reached out to the Ombi discord server for help. In the mean time I have pushed a commit that refactors how HTTP requests are made based on how the tvbd_list does it.

Reply:
@gazpachoking Do you have any recommendations? I feel like I only see the following paths forward.

1. We only support the `tmdb_id` and users will have to use a lookup plugin if they have an `imdb_id`.
2. We support `imdb_id` but we have to convert it to `tmdb_id` manually. I'm not sure how. Maybe we can invoke the [tmdb_lookup](https://flexget.com/Plugins/tmdb_lookup) for them from inside ombi_list?